### ISSUE TITLE
Contact birthdays of untouched contacts not displayed in calendar

### ISSUE THREAD

**Issue by doits** [owncloud/core#22971]
### Steps to reproduce
1. Install owncloud 9.0
2. Go to the calendar app
3. Enable contact_birthdays calendar

### Expected behaviour
Display all contact's birthday entries

### Actual behaviour
No birthdays are displayed.

When I edit the birthday of a contact, it finally gets displayed. But already existing entries are not displayed until I edit the birthday.

### Server configuration
**Operating system**: Ubuntu 15.10

**Web server:** Nginx

**Database:** Mariadb

**PHP version:** 5.6.11-1ubuntu3.1

**ownCloud version:** (see ownCloud admin page) ownCloud 9.0.0 (stable)

**Calendar version:** (see ownCloud apps page) 1.0

**Updated from an older ownCloud or fresh install:** Updated from 8.2

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**List of activated apps:**

```
Enabled:
  - activity: 2.2.1
  - calendar: 1.0
  - comments: 0.2
  - contacts: 1.0.0.0
  - dav: 0.1.5
  - documents: 0.12.0
  - encryption: 1.2.0
  - federatedfilesharing: 0.1.0
  - federation: 0.0.4
  - files: 1.4.4
  - files_external: 0.5.2
  - files_pdfviewer: 0.8
  - files_sharing: 0.9.1
  - files_texteditor: 2.1
  - files_trashbin: 0.8.0
  - files_versions: 1.2.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - gallery: 14.5.0
  - mail: 0.3.0
  - notes: 2.0.0
  - notifications: 0.2.3
  - provisioning_api: 0.4.1
  - systemtags: 0.2
  - tasks: 0.8.1
  - templateeditor: 0.1
  - updatenotification: 0.1.0
  - user_ldap: 0.8.0
Disabled:
  - external
  - user_external
```

**The content of config/config.php:**

```
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "\/var\/www\/owncloud\/data",
        "dbtype": "mysql",
        "dbname": "owncloud",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "version": "9.0.0.19",
        "installed": true,
        "forcessl": true,
        "maxZipInputSize": 838860800,
        "allowZipDownload": true,
        "theme": "",
        "log_type": "owncloud",
        "loglevel": 1,
        "log_query": false,
        "logdateformat": "Y.m.d H:i:s",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "mail_smtpmode": "php",
        "mail_from_address": "owncloud",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "http:\/\/localhost",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "forceSSLforSubdomains": true,
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "localhost",
            "port": 6379,
            "timeout": 0,
            "dbindex": 0
        },
        "logtimezone": "UTC",
        "asset-pipeline.enabled": false,
        "ldapIgnoreNamingRules": false,
        "trashbin_retention_obligation": "auto",
        "maintenance": false,
        "updatechecker": false
    }
}
```

**Are you using external storage, if yes which one:** nope

**Are you using encryption:** nope

**Are you using an external user-backend, if yes which one:** ldap, but currently logged in as as local user

#### LDAP configuration (delete this part if not used)

```
I think not needed for this issue ...
```

### Client configuration
**Browser:** Firefox 46.0a2

**Operating system:** OSX 10.11.4 beta

**CalDAV-clients:** nope

#### ownCloud log (data/owncloud.log)
```
{"reqId":"zxwRD\/jVxD7gb1SDS9HB","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:51:52"}
{"reqId":"TL4ZXI4\/uQIaHUkc+C5E","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:52:52"}
{"reqId":"umhotEbM17mkwnEjk4J0","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:52:54"}
{"reqId":"o00SWpE84drd7noxf685","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_App' not found at \/var\/www\/owncloud\/apps\/tasks\/controller\/pagecontroller.php#60","level":3,"time":"2016.03.08 18:52:56"}
{"reqId":"s16jvukDksj\/zJvmNqBU","remoteAddr":"92.228.155.152","app":"PHP","message":"Class 'OC_Calendar_Calendar' not found at \/var\/www\/owncloud\/apps\/tasks\/service\/tasksservice.php#112","level":3,"time":"2016.03.08 19:02:36"}
```

refs https://github.com/owncloud/calendar/issues/298

**Comment by LukasReschke** [comment]
cc @DeepDiver1975

**Comment by twistedfall** [comment]
You should try running:

occ dav:sync-birthday-calendar [username]

**Comment by doits** [comment]
> occ dav:sync-birthday-calendar [username]

yep, worked! Should simply be run during upgrade for all users then.

**Comment by DeepDiver1975** [comment]
We might want to add a cron job on this?

**Comment by twistedfall** [comment]
It should actually happen automatically on contact save, but I haven't really checked that.

**Comment by DeepDiver1975** [comment]
> It should actually happen automatically on contact save

this is the case - this issue is talking about untouched contacts after migration.

TBH: a migration step on this is missing :cry:

**Comment by ochurlaud** [comment]
@DeepDiver1975 is it possible to force the update without the command? (I'm on a server OVH server without terminal access). Thx

**Comment by DeepDiver1975** [comment]
Please have a look in my repos - there is the web Shell app. It will allow you to execute occ commands in the Browser.
I did not yet test it with oc9.

**Comment by lama0900** [comment]
So and what do to if already upgraded to 9.0 from 8.2.2 ? Don't say, wait for 9.0.1, always getting errors on DAVDroid b/c of this error isn't that funny.

**Comment by DeepDiver1975** [comment]
Use the occ command as explained above

**Comment by lama0900** [comment]
Oh sorry, thank you!

**Comment by DeepDiver1975** [comment]
The birthday calendar is readonly because it is one way synced from your contacts

**Comment by lama0900** [comment]
Uhm, why can I delete dates from that calendar then, and afterwards i've to sync again (with the command above)?

**Comment by DeepDiver1975** [comment]
Thats a bug in 9.0.0 which will be fixed in 9.0.1

**Comment by lama0900** [comment]
Ok. Thank you. Can be locked I think.
